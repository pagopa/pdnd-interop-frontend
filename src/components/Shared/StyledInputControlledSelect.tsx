import React, { ChangeEventHandler } from 'react'
import { InputBaseComponentProps, MenuItem, TextField } from '@mui/material'
import { SelectOption } from '../../../types'
import { StyledInputWrapper } from './StyledInputWrapper'
import { SxProps } from '@mui/system'

type StyledInputControlledSelectProps = {
  name: string
  value?: string
  error?: string
  onChange?: ChangeEventHandler
  label?: string

  disabled?: boolean
  infoLabel?: string

  inputProps?: InputBaseComponentProps
  focusOnMount?: boolean
  sx?: SxProps

  options?: Array<SelectOption>
}

export function StyledInputControlledSelect({
  label,
  disabled = false,
  infoLabel,

  name,
  value,
  onChange,
  error,

  inputProps,
  focusOnMount = false,
  sx,

  options,
}: StyledInputControlledSelectProps) {
  if (!options || Boolean(options.length === 0)) {
    return null
  }

  const hasFieldError = Boolean(error)

  return (
    <StyledInputWrapper name={name} error={error} sx={sx} infoLabel={infoLabel}>
      <TextField
        name={name}
        value={value}
        onChange={onChange}
        id={name}
        autoFocus={focusOnMount}
        select
        disabled={disabled}
        sx={{ width: '100%' }}
        variant="standard"
        label={label}
        error={hasFieldError}
        inputProps={inputProps}
      >
        {options.map((o, i) => (
          <MenuItem key={i} value={o.value}>
            {o.label}
          </MenuItem>
        ))}
      </TextField>
    </StyledInputWrapper>
  )
}
