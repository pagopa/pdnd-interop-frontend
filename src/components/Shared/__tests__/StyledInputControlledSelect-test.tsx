import React from 'react'
import renderer from 'react-test-renderer'
import { useForm } from 'react-hook-form'
import { StyledInputControlledSelect } from '../StyledInputControlledSelect'

type Option = {
  label: string
  value: string | number
}

type SelectProps = {
  options?: Array<Option>
}

function Select({ options }: SelectProps) {
  const props = { name: 'test', rules: {}, defaultValue: options && options[0].value }

  const {
    control,
    formState: { errors },
  } = useForm()

  return (
    <StyledInputControlledSelect {...props} control={control} errors={errors} options={options} />
  )
}

describe('Snapshot', () => {
  it('matches null', () => {
    const component = renderer.create(<Select />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('matches select', () => {
    const options = [
      { label: 'Opzione 1', value: 1 },
      { label: 'Opzione 2', value: 2 },
    ]
    const component = renderer.create(<Select options={options} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
