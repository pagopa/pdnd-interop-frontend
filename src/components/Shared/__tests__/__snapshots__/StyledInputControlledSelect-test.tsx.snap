// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot matches null 1`] = `null`;

exports[`Snapshot matches select 1`] = `
<div
  className="MuiBox-root css-62igne"
>
  <div
    className="MuiFormControl-root MuiTextField-root css-136h9lv-MuiFormControl-root-MuiTextField-root"
  >
    <div
      className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiSelect-root css-1480iag-MuiInputBase-root-MuiInput-root"
      onClick={[Function]}
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="test"
        className="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
        id="test"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="notranslate"
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </div>
      <input
        aria-hidden={true}
        autoFocus={false}
        className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        disabled={false}
        name="test"
        onAnimationStart={[Function]}
        onChange={[Function]}
        required={false}
        tabIndex={-1}
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
